src/segment.rs

```diff
@@ -607,10 +568,12 @@ impl Segment {
         } = self;
         let mmap_len = mmap.len();
 
-        // Unmaps the file before `fs::remove_file` else access will be denied
+        // Unmaps the file before `fs::remove_file`
+        // else access will be denied on Windows or Docker containers with Windows host
         mmap.flush()?;
         std::mem::drop(mmap);
 
+        // Do several attempts to delete the file because on Windows it may be denied
```

Comment:
Not necessarily related to this PR, but should we only retry on Windows?

Reply:
Will add a description after tests on Windows (it takes longer time that I expected). Anyway I gonna answer here.
This change is not related to https://github.com/qdrant/wal/pull/64. But I expect we can have the same problem with deletion. Because `cfg(unix)` is passed in the case on docker with Windows host (and NTFS filesystem as well). Maybe the deletion case has to be a separate PR

Reply:
And if we compare unix and windows deletion on master, we will find only 2 differences:
1. multiple attemps
2. drop memmap before file deletion
Both differences I find OK for unix case

Reply:
> is passed in the case on docker with Windows host (and NTFS filesystem as well)

Good that you mention this! Didn't think about that.

Agree, with that in mind your number 1 and 2 are prefect :ok_hand:

Reply:
It's not my idea =) It's idea from https://github.com/qdrant/wal/pull/64
This PR is just a simplification of this PR and adding deletion case

Reply:
Opened it right now to check if it fixes bug
https://github.com/qdrant/qdrant/issues/2620